### PR TITLE
Fix small return types.

### PR DESCRIPTION
**Pull request by lachlansneff**
This fixes returning integers smaller than 32bits from imported functions. Also some work on fixing returning errors from functions.

### CODE REVIEW

**Comment by lachlansneff** [comment]
bors try

**Comment by lachlansneff** [comment]
bors try

**Comment by syrusakbary** [comment]
It works on Linux, @xmclark will test on Windows

File: lib/wasi/src/syscalls/types.rs [hunk 1]
```diff
@@ -43,84 +43,84 @@ pub struct __wasi_dirent_t {
...
+pub const __WASI_EPROTOTYPE: u16 = 67;
+pub const __WASI_ERANGE: u16 = 68;
+pub const __WASI_EROFS: u16 = 69;
+pub const __WASI_ESPIPE: u16 = 70;
+pub const __WASI_ESRCH: u16 = 71;
+pub const __WASI_ESTALE: u16 = 72;
+pub const __WASI_ETIMEDOUT: u16 = 73;
+pub const __WASI_ETXTBSY: u16 = 74;
+pub const __WASI_EXDEV: u16 = 75;
+pub const __WASI_ENOTCAPABLE: u16 = 76;
```
Comment (MarkMcCaskey) [inline lib/wasi/src/syscalls/types.rs]: We should revert everything else.  The fix changed _all_ u16s to u32s

Comment (syrusakbary) [reply lib/wasi/src/syscalls/types.rs]: Fixed

**Comment by MarkMcCaskey** [comment]
bors retry

**Pull request merged**